server.go

```diff
@@ -273,43 +272,50 @@ func listAthleteActivities(oAuthToken string, startUnix int64) []Activity {
 	}
 	results := 1
 	for results > 0 {
-		req, err := http.NewRequest("GET", "https://www.strava.com/api/v3/athlete/activities"+requestQuery, strings.NewReader(""))
-		if err != nil {
-			log.Printf("error setting up new request to activities - %v", err)
-			return activities
-		}
-		defer req.Body.Close()
+		// call a function so we can use defer statements for closing .Body()
```

Comment:
the comment captures it: we had `defer` statements in a for-loop, but defers only run when the function exits. By placing this code in an in-line anonymous function, we get to keep the defer statements with minimal changing of the code. I have not verified that this did not break anything, but I don't think it should.